ecmascript/transforms/typescript/tests/strip.rs

```diff
@@ -3488,3 +3504,58 @@ to!(
     "
     "
 );
+
+test_with_config!(
+    issue_1472_1_define,
+    strip::Config {
+        use_define_for_class_fields: true,
+        ..Default::default()
+    },
+    "
+    class A extends Object {
+        a = 1;
+        constructor(public b = 2) {
+          super();
+        }
+    }
+    ",
+    "
+    class A extends Object {
+        constructor(b = 2){
+            super();
+            this.b = b;
+            Object.definedProperty(this, 'a', {
+                enumerable: true,
+                configurable: true,
+                writable: true,
+                value: 1
+            });
+        }
+    }
+    "
+);
+
+test_with_config!(
+    issue_1472_1_no_define,
+    strip::Config {
+        use_define_for_class_fields: false,
+        ..Default::default()
+    },
+    "
+    class A extends Object {
+        a = 1;
+        constructor(public b = 2) {
+          super();
+        }
+    }
+    ",
+    "
+    class A extends Object {
+        a = 1;
+        constructor(b = 2){
+            super();
+            this.b = b;
+        }
+    }
+    "
+);
```

Comment:
This is incorrect. See the examples in #1472. `use_define_for_class_fields: true` should give:
```js
class A extends Object {
    b;
    a = 1;
    constructor(b = 2){
        super();
        this.b = b;
    }
}
```
And `use_define_for_class_fields: false` should give:
```js
class A extends Object {
    constructor(b = 2){
        super();
        this.b = b;
        this.a = 1;
    }
}
```
`Object.defineProperty()` is for ES2020 and earlier.

Reply:
It's intentional. It's because `strip` does not care about the target es version.

The caller can use `use_define_for_class_fields: true` to get

```ts
class A extends Object {
    constructor(b = 2) {
        super();
        this.b = b;
        Object.defineProperty(this, "a", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: 1
        });
    }
}
```

where `this.b = b` can be transformed using another pass.


And for, `use_define_for_class_fields: false `, caller should use `class_properties` pass to convert class properties to `this.foo = value`.


But well, I guess I should add some integration tests.

Reply:
Oh, I found that this pass eats all properties, and mixing passes does not work. Thanks for pointing it out!